pandas/core/frame.py

```diff
@@ -8531,8 +8539,11 @@ def _reduce(
             labels = self._get_agg_axis(axis)
             constructor = self._constructor
 
-        def f(x):
-            return op(x, axis=axis, skipna=skipna, **kwds)
+        def func(values):
+            if is_extension_array_dtype(values.dtype):
+                return extract_array(values)._reduce(name, skipna=skipna, **kwds)
+            else:
```

Comment:
can we not use the same exact path for non extension arrays

these if/then for extension arrays are terrible

Reply:
we'd have to put the analogous check in each of the `nanops.nanfoo` functions, which I know @jorisvandenbossche wouldn't like.

Reply:
is `func` now effectively the same as `blk_func` on L8575?

Reply:
Indeed, doing it here is IMO much cleaner than in each nanops function.

> is func now effectively the same as blk_func on L8575?

Not exactly, the axis handling is different (also `blk_func` knows it gets an array, so the extract_array stuff is not needed).

Reply:
I guess this is ok as a short term check, but seeing these constant if is_extension_array do one thing, else do something else basically means the api is missing lots of things. I would *much* rather fix it than keep doing this kind of thing. This makes code extremly hard to read and understand.

I remember being very excited when we removed the is_ndarray check here and this because much simpler, now we are going backwards.